dipy/align/transforms.pyx

```diff
@@ -0,0 +1,955 @@
+    def __init__(self):
+        r""" Translation transform in 3D
+        """
+        self.dim = 3
+        self.number_of_parameters = 3
+
+    cdef int _jacobian(self, double[:] theta, double[:] x, double[:,:] J) nogil:
+        r""" Jacobian matrix of the 3D translation transform
+        The transformation is given by:
+
+        T(x) = (T1(x), T2(x), T3(x)) = (x0 + t0, x1 + t1, x2 + t2)
+
+        The derivative w.r.t. t1, t2 and t3 is given by
+
+        T'(x) = [[1, 0, 0], # derivatives of [T1, T2, T3] w.r.t. t0
+                 [0, 1, 0], # derivatives of [T1, T2, T3] w.r.t. t1
+                 [0, 0, 1]] # derivatives of [T1, T2, T3] w.r.t. t2
+        Parameters
+        ----------
+        theta : array, shape(3,)
+            the parameters of the 3D translation transform (the Jacobian does not
+            depend on the parameters, but we receive the buffer so all Jacobian
+            functions receive the same parameters)
+        x : array, shape(3,)
+            the point at which to compute the Jacobian (the Jacobian does not
+            depend on x, but we receive the buffer so all Jacobian functions
+            receive the same parameters)
+        J : array, shape(3, 3)
+            the buffer in which to write the Jacobian
+
+        Returns
+        -------
+        is_constant : int
+            always returns 1, indicating that the Jacobian is constant
+            (independent of x)
+        """
+        J[0,0], J[0,1], J[0,2] = 1.0, 0.0, 0.0
+        J[1,0], J[1,1], J[1,2] = 0.0, 1.0, 0.0
+        J[2,0], J[2,1], J[2,2] = 0.0, 0.0, 1.0
+        # This Jacobian does not depend on x (it's constant): return 1
+        return 1
+
+    cdef void _get_identity_parameters(self, double[:] theta) nogil:
+        r""" Parameter values corresponding to the identity
+        Sets in theta the parameter values corresponding to the identity
+        transform
+
+        Parameters
+        ----------
+        theta : array, shape(3,)
+            buffer to write the parameters of the 3D translation transform
+        """
+        theta[:3] = 0
+
+    cdef void _param_to_matrix(self, double[:] theta, double[:,:] R) nogil:
+        r""" Matrix associated with the 3D translation transform
+
+        Parameters
+        ----------
+        theta : array, shape(3,)
+            the parameters of the 3D translation transform
+        R : array, shape(4, 4)
+            the buffer in which to write the translation matrix
+        """
+        R[0,0], R[0,1], R[0,2], R[0,3] = 1, 0, 0, theta[0]
+        R[1,0], R[1,1], R[1,2], R[1,3] = 0, 1, 0, theta[1]
+        R[2,0], R[2,1], R[2,2], R[2,3] = 0, 0, 1, theta[2]
+        R[3,0], R[3,1], R[3,2], R[3,3] = 0, 0, 0, 1
+
+
+cdef class RotationTransform2D(Transform):
+    def __init__(self):
+        r""" Rotation transform in 2D
+        """
+        self.dim = 2
+        self.number_of_parameters = 1
+
+    cdef int _jacobian(self, double[:] theta, double[:] x, double[:,:] J) nogil:
```

Comment:
Sure!, I did a full style check. Is it ok to remove the spaces when it improves readability, like this?:
https://github.com/nipy/dipy/pull/574/files#diff-7e2c30721bedfedcee9a247ccfc88934R422
If I add spaces I would need to break the line, but the code is much more readable as it is (the matrix structure is easily seen)